Coroutine.ttslua

```diff
@@ -1,16 +1,28 @@
 ---@class ge_tts__Coroutine
 local Coroutine = {}
 
---- Yields from the current coroutine, which will later be resumed once a condition is met.
+--- Yields from the current coroutine, to be resumed once a condition is met or an optional timeout limit is reached.
 ---@param condition fun(): boolean @Return true when the current coroutine should be resumed.
-function Coroutine.yieldCondition(condition)
+---@param timeout number|nil @Timeout in seconds (optional).
+---@return boolean @Return whether the condition was met or not.
+---@overload fun(condition: fun():boolean):boolean
```

Comment:
I must admit, I've been inconsistent with this myself, however, I'd prefer spaces to follow `:`.

Also, if you could put `@overload`s at the top (above `@params` and `@return`, but below the description), that'd be appreciated, as it makes overloads consistently discoverable. Aside from consistency, it'd good to ensure overloads are prominent as they tend to be used more often than the main signature.

However, the return type of this overload should actually be `true`, not `boolean` i.e.

```lua
---@overload fun(condition fun(): boolean): true
```

The reason for this is that this function will _never_ return `false` if the `timeout` parameter is omitted. This is important information, as it lets consumers know they can safely ignore the return value _unless_ they're providing a `timeout`.